cmd/kubeadm/app/phases/kubelet/flags_test.go

```diff
@@ -200,14 +188,14 @@ func TestReadKubeadmFlags(t *testing.T) {
 	}{
 		{
 			name:          "valid kubeadm flags with container-runtime-endpoint",
-			fileContent:   `KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock --pod-infra-container-image=registry.k8s.io/pause:1.0"`,
-			expectedValue: []string{"--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock", "--pod-infra-container-image=registry.k8s.io/pause:1.0"},
+			fileContent:   `KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock"`,
+			expectedValue: []string{"--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock"},
```

Comment:
we need to read the file on upgrade remove the flag and write it again. we have utils for that iirc

Reply:
ok, i will update it later.

Reply:
Should I do this in the `post-upgrade` phase or create a new phase?

For init and join commands, they write the kubelet dynamic environment flags file in the `kubelet-start` phase.